src/main/app/controller/courts/LocalAuthoritiesController.ts

```diff
@@ -0,0 +1,193 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {CourtType} from '../../../types/CourtType';
+import {
+  LocalAuthoritiesAreaOfLaw,
+  LocalAuthoritiesPageData,
+  LocalAuthority,
+  LocalAuthorityItem
+} from '../../../types/LocalAuthority';
+import {AreaOfLaw} from '../../../types/AreaOfLaw';
+
+import {SelectItem} from '../../../types/CourtPageData';
+import {CSRF} from '../../../modules/csrf';
+
+export enum familyAreaOfLaw {
+  adoption = 'Adoption',
+  children = 'Children',
+  civilPartnership = 'Civil partnership',
+  divorce = 'Divorce'
+}
+@autobind
+export class LocalAuthoritiesController {
+
+
+  getCourtTypesErrorMsg = 'A problem occurred when retrieving the court types.';
+  getCourtAreasOfLawErrorMsg = 'A problem occurred when retrieving the court areas of law. ';
+  getLocalAuthoritiesErrorMsg = 'A problem occurred when retrieving the list of local authorities. ';
+  getCourtLocalAuthoritiesErrorMsg = 'A problem occurred when retrieving the court local authorities. ';
+  updateErrorMsg = 'A problem occurred when saving the local authorities. ';
+  familyAreaOfLawErrorMsg = 'You need to enable relevant family court areas of law ';
+
+  public async getAreasOfLaw(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    areasOfLaw: AreaOfLaw[] = null): Promise<void> {
+    const slug: string = req.params.slug as string;
+
+    if (!areasOfLaw ) {
+      await req.scope.cradle.api.getCourtAreasOfLaw(slug)
+        .then((value: AreaOfLaw[]) => areasOfLaw = value)
+        .catch(() => error += this.getCourtAreasOfLawErrorMsg);
+    }
+
+    if (areasOfLaw ){
+      areasOfLaw = this.checkFamilyAreasOfLaw(areasOfLaw);
+      if(!areasOfLaw.length){
+        error += this.familyAreaOfLawErrorMsg;
+      }
+    }
+
+    let courtTypes: CourtType[] = [];
+    await req.scope.cradle.api.getCourtCourtTypes(slug)
+      .then((value: CourtType[]) => courtTypes = value)
+      .catch(() => error += this.getCourtTypesErrorMsg);
+
+    const pageData: LocalAuthoritiesAreaOfLaw = {
+      errorMsg: error,
+      updated: updated,
+      isEnabled: !courtTypes.length ? false :courtTypes.some(c => c.name === 'Family Court'),
+      courtAreasOfLaw: this.getAreasOfLawForSelect(areasOfLaw, ''),
+    };
+
+    res.render('courts/tabs/localAuthoritiesContent', pageData);
+  }
+
+
+  public async getLocalAuthorities(
+    req: AuthedRequest,
+    res: Response,
+    updated = false,
+    error = '',
+    selectedAreaOfLaw = '',
+    courtLocalAuthorities: LocalAuthority[] = null): Promise<void> {
+    const slug: string = req.params.slug as string;
+    const areaOfLaw: string = req.params.areaOfLaw as string;
+
+    let areasOfLaw: AreaOfLaw[] = null;
```

Comment:
This confuses me a little. By declaring: 

`let areasOfLaw: AreaOfLaw[] = null;` - it's always null on declaration 

```
    if (!areasOfLaw ) { <- always null by default as declared above, so goes in here 

        set areas of law 

    }

    if (areasOfLaw ){
         areas of law is always set, unless there is an error
    }
```

Is it better to do something like: 

```
let areasOfLaw: AreaOfLaw[] = null;
await req.scope.cradle.api.getCourtAreasOfLaw(slug)
        .then((value: AreaOfLaw[]) => areasOfLaw = value)
        .catch(() => {

        error += this.getCourtAreasOfLawErrorMsg;
        areasOfLaw = this.checkFamilyAreasOfLaw(areasOfLaw);


} );

```

Reply:
refactored code as your suggestion